Compile error with get_drive function for android template

**Operating system or device, Godot version, GPU Model and driver (if graphics related):**
Ubuntu 16.04.3, master branch 3f4062c

**Issue description:**
<!-- What happened, and what was expected. -->
it's from #11160.
I got compile error for building android template.

```
drivers/unix/dir_access_unix.cpp:209:15: error: use of undeclared identifier 'setmntent'
        FILE *mtab = setmntent("/etc/mtab", "r");
                     ^
drivers/unix/dir_access_unix.cpp:214:10: error: use of undeclared identifier 'getmntent_r'
                while (getmntent_r(mtab, &mnt, strings, sizeof(strings))) {
                       ^
drivers/unix/dir_access_unix.cpp:223:3: error: use of undeclared identifier 'endmntent'; did you mean 'getmntent'?
                endmntent(mtab);
                ^~~~~~~~~
                getmntent
/mnt/2TB/Development/NVPACK/android-ndk-r15c-linux/platforms/android-18/arch-arm/usr/include/mntent.h:49:16: note: 
      'getmntent' declared here
struct mntent* getmntent(FILE*);
               ^
3 errors generated.
scons: *** [drivers/unix/dir_access_unix.android.opt.armv7.neon.o] Error 1
```

this is what `/android-ndk-r15c-linux/platforms/android-18/arch-arm/usr/include/mntent.h` has inside
```cpp
#ifndef _MNTENT_H_
#define _MNTENT_H_

#include <stdio.h>

#define MNTTYPE_IGNORE "ignore"

struct mntent
{
    char* mnt_fsname;
    char* mnt_dir;
    char* mnt_type;
    char* mnt_opts;
    int mnt_freq;
    int mnt_passno;
};

__BEGIN_DECLS
struct mntent* getmntent(FILE*);
__END_DECLS

#endif
```

I found `/android-ndk-r15c-linux/platforms/android-21/arch-arm/usr/include/mntent.h` has those functions.
```cpp
int endmntent(FILE*);
struct mntent* getmntent(FILE*);
struct mntent* getmntent_r(FILE*, struct mntent*, char*, int);
FILE* setmntent(const char*, const char*);
```

**Steps to reproduce:**
`scons platform=android tools=no`

**Link to minimal example project:**
<!-- Optional but very welcome. You can drag and drop a zip archive to upload it. -->

I compiled `drivers/unix/dir_access_unix.cpp` successfully with `scons platform=android tools=no ndk_platform=android-21`
maybe we should raise default version from here?
https://github.com/godotengine/godot/blob/master/platform/android/detect.py#L24
@RandomShaper what do you think?

it's out of this issue, but I got another error. :(
```
thirdparty/recastnavigation/Recast/Source/Recast.cpp:26:10: fatal error: 'new' file not found
#include <new>
         ^~~~~
1 error generated.
```

--- a/drivers/unix/dir_access_unix.cpp
+++ b/drivers/unix/dir_access_unix.cpp
@@ -183,17 +183,18 @@ void DirAccessUnix::list_dir_end() {
 	_cisdir = false;
 }
 
-#ifdef HAVE_MNTENT
+#if defined(HAVE_MNTENT) && defined(X11_ENABLED)
 static bool _filter_drive(struct mntent *mnt) {
 	// Ignore devices that don't point to /dev
 	if (strncmp(mnt->mnt_fsname, "/dev", 4) != 0) {
 		return false;
 	}
 
-	// Accept devices mounted at /media, /mnt or /home
+	// Accept devices mounted at common locations
 	if (strncmp(mnt->mnt_dir, "/media", 6) == 0 ||
 			strncmp(mnt->mnt_dir, "/mnt", 4) == 0 ||
-			strncmp(mnt->mnt_dir, "/home", 5) == 0) {
+			strncmp(mnt->mnt_dir, "/home", 5) == 0 ||
+			strncmp(mnt->mnt_dir, "/run/media", 10) == 0) {
 		return true;
 	}
 
@@ -204,7 +205,7 @@ static bool _filter_drive(struct mntent *mnt) {
 
 static void _get_drives(List<String> *list) {
 
-#ifdef HAVE_MNTENT
+#if defined(HAVE_MNTENT) && defined(X11_ENABLED)
 	// Check /etc/mtab for the list of mounted partitions
 	FILE *mtab = setmntent("/etc/mtab", "r");
 	if (mtab) {

